front-end/src/components/Absence/Absence.vue

```diff
@@ -449,6 +517,45 @@ export default {
       this.editAbsenceFullname = fn;
       this.dialogVisible = true;
     },
+    onAttachmentDownload(file) {
+      window.open(file, "_blank");
+    },
+    onAttachmentUpload(id) {
+      this.$refs.attachmentUpload.value = null;
+      this.uploadAttachmentAbsenceId = id;
+      this.$refs.attachmentUpload.click();
+    },
+    onAttachmentPicked() {
+      const formData = new FormData();
+      if (this.$refs.attachmentUpload.files[0]) {
+        formData.set("image", this.$refs.attachmentUpload.files[0]);
+      }
+      patchAbsence(this.uploadAttachmentAbsenceId, formData)
+        .then(() => {
+          patchSuccess("приложения");
+          this.dialogVisible = false;
+          this.onFilter();
+        })
+        .catch(err => patchError("приложения", err.response.status));
```

Comment:
Use `async` / `await`.

Reply:
Fixed. But I'm not sure that I got u right. Check please